fairlearn/datasets/_fetch_compas.py

```diff
@@ -0,0 +1,139 @@
+    race_African-American  1 if defendant's race African-American, otherwise 0
+    race_Caucasian         1 if defendant's race Caucasian, otherwise 0
+    c_charge_degree_F      1 if charged for felony, otherwise 0
+    c_charge_degree_M      1 if charged for misdemeanor, otherwise 0
+    =====================  ======================================================================
+
+    Parameters
+    ----------
+    cache : boolean, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : optional, default: None
+        Specify another download and cache folder for the datasets. By default
+        all fairlearn data is stored in '~/.fairlearn-data' subfolders.
+
+    as_frame : boolean, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string or categorical). The target is
+        a pandas DataFrame or Series depending on the number of target_columns.
+        The Bunch will contain a ``frame`` attribute with the target and the
+        data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
+        DataFrames or Series as describe above.
+
+    return_X_y : boolean, default=False.
+        If True, returns ``(data.data, data.target)`` instead of a Bunch
+        object.
+
+    warn : boolean, default=True.
+        If True, it raises an extra warning to make users aware of the unfairness
+        aspect of this dataset.
+
+
+    Returns
+    -------
+    dataset : :class:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the following attributes.
+
+        data : ndarray, shape (5278, 13)
+            Each row corresponding to the 13 feature values in order.
+            If ``as_frame`` is True, ``data`` is a pandas object.
+        target : numpy array of shape (5278,)
+            Each value indicates whether a defendant was charged again within two years after the
+            current charge.
+            If ``as_frame`` is True, ``target`` is a pandas object.
+        feature_names : list of length 13
+            Array of ordered feature names used in the dataset.
+        DESCR : string
+            Description of the COMPAS dataset.
+
+    (data, target) : tuple of (numpy.ndarray, numpy.ndarray) or (pandas.DataFrame, pandas.Series)
+        if ``return_X_y`` is True and ``as_frame`` is False
+
+    (data, target) : tuple of (pandas.DataFrame, pandas.Series)
+        if ``return_X_y`` is True and ``as_frame`` is True
+
+    See Also
+    --------
+    fairlearn.datasets.fetch_boston: Boston housing dataset (regression)
+    fairlearn.datasets.fetch_adult: UCI Adult dataset (binary classification)
+
+    Notes
+    -----
+    This dataset consists of 5278 samples and 13 features.
+
+    - During pre-processing only instances of defendants of the two most common races were kept,
+      which means that `race_African-American` and `race_Caucasian` are perfectly correlated.
```

Comment:
What do you mean by correlated? The first half sentence seems to indicate that it's just two races, so I don't know what that would correlate with.

Reply:
I guess my phrasing is not very clear. Because race_African-American and race-Caucasian are the only two possible races in the dataset, these two features have a perfect negative correlation (i.e. -1). Same for c_charge_degreeF and c_charge_degree_M. 

The reason I wrote this down explicitly is that including the redundant features influences the interpretation as well as the stability of the model you build. E.g. AFAIK in linear regression highly correlated features will mess up the variance of the weights resulting in an unstable model.

Alternatively, we can just remove the redundant features from the dataset and very clearly write down what the remaining variables mean; e.g. call it 'race' and explain that 1 means African-American and 0 Caucasian.

Reply:
Got it! I somehow interpreted this as `race_African-American` being one value of the race column and `race-Caucasian` being another, but they're columns themselves with binary values. This makes sense. If you want you can add that disclaimer about redundancy. I think it's useful!

Reply:
Given the instability you mentioned, is there a particular reason for keeping these columns (rather than a single `race` column)? I agree that 'redundant' makes more sense than 'correlated' in this context.

Reply:
One reason may be to avoid confusion with what the column means if people don't read the documentation. But I'm also on board with removing one of them.

Reply:
I'm sure people will manage to confuse themselves with or without documentation :-)

I suppose that one advantage of keeping this 'one-hot-encoding' model even for the binary feature is that it avoids the need to decide which group gets 0 and which gets 1. I'd like to understand that potential instability a little better; I am concerned that if the dataset is tricky to train a model on in the first place, users could get lost in those weeds, and think of the fairness issues in those terms, rather than the data themselves containing bias.

Reply:
It's not a huge issue necessarily. But what will happen is that for most ML models the usage of either `race_African-American`, `race-Caucasian`, or 'a little bit of both' will be arbitrary. Especially in the latter case, this might lead to confusion when interpreting the importance of features, e.g. in terms of the coefficients of a logistic regression model.

Reply:
> If the idea is to use this for examples or educational materials then it would be great to hear thoughts on that (particularly in terms of a [real deployment context and real harms](https://fairlearn.github.io/contributor_guide/contributing_example_notebooks.html)). It's already super exciting to see the discussion here about escaping the framing trap that comes with focusing on the dataset itself :) There are highlights from several other references in #478 that might be helpful.
> 
> As with the Boston housing dataset, if other folks disagree that's cool and I'll step back and let you all decide what makes sense from here. Thanks for listening! 😄
> 
> EDIT: I also just saw #513 too, which says:
> 
> > Removing this accordingly until we
> > (a) figure out how to capture the sociotechnical context properly OR
> > (b) conclude that this is an area where AI systems should not be used at all and document that conclusion.
> > We should definitely avoid the solutionism trap from Selbst et al.'s paper "Fairness and Abstraction in Sociotechnical Systems".
> 
> Maybe that discussion has already happened offline?

@kevinrobinson there were no further discussions. @hildeweerts is just adding datasets, that's all :-)
Just didn't feel very good about having that notebook there, and the fact that people seem to be taking notebooks as best practices makes me feel very good about having removed it.

Reply:
My suggestion: let's keep both columns, but update the description:
* perfectly correlated -> perfectly anticorrelated